src/blocks/extensions/popup-builder/edit.js

```diff
@@ -0,0 +1,172 @@
+			variations:
+			typeof getBlockVariations === 'undefined'
+				? null
+				: getBlockVariations( name ),
+			isParentOfSelectedBlock: select( 'core/block-editor' ).hasSelectedInnerBlock( clientId, true ),
+		};
+	} );
+
+	const { replaceInnerBlocks } = useDispatch( 'core/block-editor' );
+
+	// Add and remove the CSS on the drop and remove of the component.
+	useLayoutEffect( () => {
+		styles.use();
+		return () => {
+			styles.unuse();
+		};
+	}, [] );
+
+	useEffect( () => {
+		// Assigning block_id in the attribute.
+		setAttributes( { block_id: clientId.substr( 0, 8 ) } );
+	}, [] );
+
+	useEffect( () => {
+		// Replacement for componentDidUpdate.
+		const blockStyling = styling( props );
+        addBlockEditorDynamicStyles( 'uagb-popup-builder-style-' + clientId.substr( 0, 8 ), blockStyling );
+	}, [ attributes, deviceType ] );
+
+	if ( isParentOfSelectedBlock ) {
+		const emptyBlockInserter = document.querySelector( '.block-editor-block-list__empty-block-inserter' );
+		if ( emptyBlockInserter ) {
+			emptyBlockInserter.style.display = 'none';
+		}
+	}
+
+	const createBlocksFromInnerBlocksTemplate = ( innerBlocksTemplate ) => {
+		return innerBlocksTemplate.map(
+			( [ name, attributes, innerBlocks = [] ] ) => ( // eslint-disable-line no-shadow
+				createBlock(
+					name,
+					attributes,
+					createBlocksFromInnerBlocksTemplate( innerBlocks )
+				)
+			)
+		);
+	};
+
+	const blockVariationPickerOnSelect = (
+		nextVariation = defaultVariation
+	) => {
+		if ( nextVariation.attributes ) {
+			setAttributes( nextVariation.attributes );
+		}
+
+		if ( nextVariation.innerBlocks ) {
+			replaceInnerBlocks(
+				clientId,
+				createBlocksFromInnerBlocksTemplate( nextVariation.innerBlocks )
+			);
+		}
+	};
+
+	const postType = useSelect( ( select ) => select( 'core/editor' ).getCurrentPostType() ); // eslint-disable-line no-shadow
+	const [ meta, setMeta ] = useEntityProp( 'postType', postType, 'meta' );
+
+	useEffect( () => {
+		if ( 'spectra-popup' !== postType ) {
+			return;
+		}
+		setMeta(
+			( ! variationSelected ) ? {
+				...meta,
+				'spectra-popup-type': 'unset',
+				'spectra-popup-enabled': 'false',
+			} : {
+				...meta,
+				'spectra-popup-type': variantType,
+			}
+		);
+	}, [
+		variationSelected,
+		variantType,
+	] );
+
+
+	if ( ! variationSelected && 0 === select( 'core/block-editor' ).getBlockParents( clientId ).length ) {
+		return (
+			<div className='uagb-popup-builder__variation-picker'>
+				<BlockVariationPicker
+					icon={ '' }
+					label={ __(
+						'Select a Popup Type',
+						'ultimate-addons-for-gutenberg'
+					) }
+					instructions={ false }
+					variations={ variations }
+					onSelect={ ( nextVariation ) =>
+						blockVariationPickerOnSelect( nextVariation )
+					}
+				/>
+				{/* Add a Dropdown to Select an Already Existing Popup */}
+			</div>
+		);
+	}
+
+	return (
+		<>
+			{ isSelected && <Settings { ...{ ...props } } /> }
+			<Render { ...{ ...props } } />
```

Comment:
```suggestion
			{ isSelected && <Settings { ...props }  /> }
			<Render { ...props }  />
```
Can you use this way

Reply:
Implemented.